### PR TITLE
[Runtimes] Support configuring specific images that should be enriched with registry

### PR DESCRIPTION
**Pull request by Hedingber**
Implements https://jira.iguazeng.com/browse/ML-337

### CODE REVIEW

File: mlrun/config.py [hunk 1]
```diff
@@ -50,6 +50,10 @@
     "version": "",  # will be set to current version
     "images_tag": "",  # tag to use with mlrun images e.g. mlrun/mlrun (defaults to version)
     "images_registry": "",  # registry to use with mlrun images e.g. quay.io/ (defaults to empty, for dockerhub)
+    # comma separated list of images that are in the specified images_registry, and therefore will be enriched with this
+    # registry when used. default to empty which means any image which is of the mlrun repository (mlrun/mlrun,
+    # mlrun/ml-base, etc...)
```
Comment (omesser) [inline mlrun/config.py:55]: empty == enrich mlrun/*
means there's no way to configure it to never enrich
use something like `mlrun/*` as default to have flexibility to have it never enrich as well when it's needed (open source?)

**Review by omesser — CHANGES_REQUESTED** [review]

File: mlrun/config.py (hunk 1 above)
Comment (Hedingber) [reply mlrun/config.py:55]: There is a way to never enrich - not configuring the `images_registry`
This config value is relevant only when enriched is wanted - i.e. `images_registry` has a value

Comment (omesser) [reply mlrun/config.py:55]: That's not desirable - can't disable an enrichment by configuration, but just use the value to make it ineffectual

File: mlrun/utils/helpers.py [hunk 2]
```diff
@@ -519,11 +519,33 @@ def _set_artifact_path(task):
...
+            for image in images_to_enrich_registry:
+                if image_url.startswith(image):
+                    enrich_registry = True
+        if enrich_registry:
             image_url = f"{registry}{image_url}"
+
+    if is_mlrun_image and tag and ":" not in image_url:
+        image_url = f"{image_url}:{tag}"
+
     return image_url
```
Comment (omesser) [inline mlrun/utils/helpers.py:549]:
```py
def enrich_image_url(image_url: str) -> str:
    tag = config.images_tag or mlrun.utils.version.Version().get()["version"]
    registry = config.images_registry

    enrich_registry = False
    # enrich registry only if images_to_enrich_registry provided
    # example: "$mlrun/*" means enrich only if the image repository is mlrun and registry is not specified (in which case
    # /mlrun/ will be part of the url)

    if config.images_to_enrich_registry:
        for pattern_to_enrich in config.images_to_enrich_registry.split(","):
          if re.match(config.images_to_enrich_registry, image_url):
            enrich_registry = True
    if enrich_registry:
      image_url = f"{registry}{image_url}"

    if is_mlrun_image and tag and ":" not in image_url:
        image_url = f"{image_url}:{tag}"
```

**Review by omesser — CHANGES_REQUESTED** [review]
suggestion inside. will require default value change

**Pull request merged**